version: '3.8'

services:
  web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    network_mode: "host"
    container_name: genuinstagram_backend
    volumes:
      - .:/genuinstagram_backend
      # - .:/genuinsta_backend
    ports:
      - 8000:8000
      - 7913:7913
    depends_on:
      - db
      - redis
    
  db:
    image: postgres:12.0-alpine
    restart: always
    network_mode: "host"
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=piyushnh
      - POSTGRES_PASSWORD=piyush1235
      - POSTGRES_DB=genuinsta
  redis:
    # container_name: 'localhost'
    image: redis:6.0.5-alpine
    network_mode: "host"
    restart: always
    ports:
    - "0.0.0.0:6379:6379"
    volumes:
      - 'redis_data:/data'
  celery:
    build: .
    command: python manage.py celery
    restart: always
    network_mode: "host"
    environment:
      - C_FORCE_ROOT=true
    volumes:
      - .:/genuinstagram_backend
      # - .:/genuinsta_backend
    depends_on:
      # - web
      - db
      - redis
    
  

volumes:
  postgres_data:
  redis_data:
    